### ISSUE TITLE
IndexOutOfBounds on setPlayWhenReady using IMA Extension

### ISSUE THREAD

**Issue by wemped** [google/ExoPlayer#4096]
### Issue description
I am receiving a significant amount of crash reports when doing `setPlayWhenReady(true)`. From the stacktrace it looks like it might have to do with the IMA Extension, but I've yet to be able to reproduce it locally.
I'm seeing this issue on all devices and android versions 5-8.

### Reproduction steps
Unable to reproduce it locally, but I can give my setup (Kotlin).
```
class VideoPlayer(...): SimpleExoPlayer(...) {
  fun play(video: VideoModel) {
    var mediaSource = mediaSourceFromVideo(video) ?: return

    if (AdHelper.shouldShowAds) {
        mediaSource = requestAds(mediaSource, video)
    }

    prepare(mediaSource)
    playWhenReady = true // <---- CRASH COMES FROM THIS CALL
  }

  private fun mediaSourceFromVideo(video: VideoModel): MediaSource? {
    val mediaSource = when {
        video.assetUrlString.isHlsUrl -> HlsMediaSource(Uri.parse(video.assetUrlString), dataSourceFactory, 1, null, null)
        video.assetUrlString.isMp4Url -> {
            val mp4Uri = Uri.parse(video.assetUrlString)
            ExtractorMediaSource(mp4Uri, dataSourceFactory, extractorsFactory, null, null)
        }
        else -> return null
    }

    if (video.closedCaptionUrlString != null) {
        val ccUri = Uri.parse(video.closedCaptionUrlString)
        val textFormat = Format.createTextSampleFormat(null, MimeTypes.TEXT_VTT, Format.NO_VALUE, "en")
        val textMediaSource = SingleSampleMediaSource(ccUri, dataSourceFactory,
                textFormat, C.TIME_UNSET)
        return MergingMediaSource(mediaSource, textMediaSource)
    }

    return mediaSource
  }

  private fun requestAds(originalMediaSource: MediaSource, video: VideoModel): MediaSource {
      val adTagUri = AdHelper.getAdTag(video) ?: return originalMediaSource
      adsLoader = ImaAdsLoader(adContainerView.context, adTagUri)

      return ImaAdsMediaSource(
              originalMediaSource,
              dataSourceFactory,
              adsLoader,
              adContainerView,
              Handler(Looper.getMainLooper()),
              this)
  } 
}
```

### Version of ExoPlayer being used
compile 'com.google.android.exoplayer:exoplayer:r2.5.2'
compile 'com.google.android.exoplayer:extension-ima:r2.5.2'

### Device(s) and version(s) of Android being used
All devices, reports coming from android version 5,6,7,8.

### A full bug report captured from the device
From Crashlytics :
```
Fatal Exception: java.lang.IndexOutOfBoundsException
       at com.google.android.exoplayer2.Timeline$1.getPeriod(Timeline.java:532)
       at com.google.android.exoplayer2.Timeline.getPeriod(Timeline.java:687)
       at com.google.android.exoplayer2.ExoPlayerImpl.getContentPosition(ExoPlayerImpl.java:375)
       at com.google.android.exoplayer2.ext.ima.ImaAdsLoader.checkForContentComplete(ImaAdsLoader.java:697)
       at com.google.android.exoplayer2.ext.ima.ImaAdsLoader.onPlayerStateChanged(ImaAdsLoader.java:556)
       at com.google.android.exoplayer2.ExoPlayerImpl.setPlayWhenReady(ExoPlayerImpl.java:166)
       at com.google.android.exoplayer2.SimpleExoPlayer.setPlayWhenReady(SimpleExoPlayer.java:627)
       at *MyApp*.video.VideoPlayer.setPlayWhenReady(VideoPlayer.kt:244)
       at *MyApp*.video.VideoPlayer.play(VideoPlayer.kt:150)
       at *MyApp*.video.VideoManager.goToInlineAndPlay(VideoManager.kt:113)
       at *MyApp*.video.VideoManager.goToInlineAndPlay$default(VideoManager.kt:107)
       at *MyApp*.feed.models.ContentFeedModel.selectedItem(ContentFeedModel.kt:246)
       at *MyApp*.feed.FeedController$FeedAdapter$onCreateViewHolder$1.onClick(FeedController.kt:142)
       at android.view.View.performClick(View.java:5637)
       at android.view.View$PerformClick.run(View.java:22433)
       at android.os.Handler.handleCallback(Handler.java:751)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:179)
       at android.app.ActivityThread.main(ActivityThread.java:6152)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
```

**Comment by ojw28** [comment]
There have been a lot of robustness fixes to the IMA extension since `2.5.2`. It's pretty likely the issue is fixed in the latest release. Please update and file an issue if the problem persists. Thanks!

**Comment by andrewlewis** [comment]
This is similar to #3334 and was likely fixed in 2c10e6f1c3fd1692d8db3d16355648aab92e66f7.
